pkg/providers/vsphere/vmprovider.go

```diff
@@ -682,16 +651,25 @@ func (vs *vSphereVMProvider) GetTasksByActID(ctx context.Context, actID string)
 	for {
 		nextTasks, err := collector.ReadNextTasks(ctx, taskHistoryCollectorPageSize)
 		if err != nil {
-			pkglog.FromContextOrDefault(ctx).Error(err, "failed to read next tasks")
+			logger.Error(err, "failed to read next tasks")
```

Comment:
Also not your change - can we just return a wrapped error instead of logging twice (here and the by the controller at the end of reconciliation)?